cmd/buildctl/common/common.go

```diff
@@ -18,6 +18,30 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
+// ResolveTLSFilesFromDir scans a TLS directory for known cert/key filenames.
+func ResolveTLSFilesFromDir(tlsDir string) (caCert, cert, key string, err error) {
+	oneOf := func(either, or string) (string, error) {
+		for _, name := range []string{either, or} {
+			fpath := filepath.Join(tlsDir, name)
+			if _, err := os.Stat(fpath); err == nil {
+				return fpath, nil
+			}
+		}
+		return "", errors.Errorf("directory did not contain one of the needed files: %s or %s", either, or)
```

Comment:
Might be better to capture actual error if the error was not `IsNotExist()`. Eg. could be that the file exists but is returning eperm.

Reply:
Done.